api/src/modules/scenario-interventions/scenario-interventions.service.ts

```diff
@@ -443,7 +451,53 @@ export class ScenarioInterventionsService extends AppBaseService<
     scenarioIntervention.newLocationType = dto.newLocationType;
     scenarioIntervention.newLocationCountryInput = dto.newLocationCountryInput;
     scenarioIntervention.newLocationAddressInput = dto.newLocationAddressInput;
+    scenarioIntervention.createDto = initialDto as unknown as JSON;
 
     return scenarioIntervention;
   }
+
+  async updateIntervention(
+    id: string,
+    dto: UpdateScenarioInterventionDto,
+  ): Promise<ScenarioIntervention> {
+    const currentScenarioIntervention: ScenarioIntervention =
+      await this.repository.findOneOrFail({ id });
+
+    // If intervention was created before breaking changes,
+    // use previous update method
+    if (!currentScenarioIntervention.createDto) {
+      return await this.update(id, dto);
+    }
+
+    for (const k of Object.keys(dto)) {
+      if (!this.basicUpdateColumns.includes(k)) {
+        return await this.replaceScenarioIntervention(id, dto);
+      }
+    }
+    return await this.update(id, dto);
+  }
+
+  /**
+   * This method is used when we need to update intervention in a way that will cause recalculation.
+   * New Scenario intervention is created and old one with all related data is deleted
+   */
+  async replaceScenarioIntervention(
+    id: string,
+    dto: UpdateScenarioInterventionDto,
+  ): Promise<ScenarioIntervention> {
+    const currentScenarioIntervention: ScenarioIntervention =
+      await this.repository.findOneOrFail({ id });
+    const createScenarioDto: CreateScenarioInterventionDto = {
+      ...(currentScenarioIntervention.createDto as unknown as CreateScenarioInterventionDto),
+      ...dto,
+    };
+    const newScenario: ScenarioIntervention =
```

Comment:
Just a small naming comment - I think here it should be newScenarioIntervention, not newScenario - since Scenarios are different, more general, entities, that may have multiples ScenarioInterventions